test/massive/munit/async/ChainedAsyncTest.hx

```diff
@@ -0,0 +1,211 @@
+ * </p>
+ */
+class ChainedAsyncTest
+{
+    static inline var L1_RESPONS_FACTOR       =  11;
+    static inline var L2_RESPONS_FACTOR       =  13;
+    static inline var L1_TIMEOUT_FACTOR       =  17;
+    static inline var L2_TIMEOUT_FACTOR       =  19;
+    static inline var FAILURE_TIMEOUT_FACTOR  =  97;
+
+    private var actualValue   : Int;
+    private var expectedValue : Int;
+
+    private var h1     : Void->Void;
+    private var h2     : Void->Void;
+    private var h3     : Void->Void;
+    private var h4     : Void->Void;
+    private var h5     : Void->Void;
+    private var t1     : Void->Void;
+    private var t2     : Void->Void;
+    private var t3     : Void->Void;
+    private var t4     : Void->Void;
+    private var t5     : Void->Void;
+
+    public function new()
+    {
+    }
+
+    @Before
+    public function setup():Void
+    {
+        actualValue = 1;
+    }
+
+    @After
+    public function tearDown():Void
+    {
+        Assert.areEqual (expectedValue, actualValue);
+    }
+
+    /**
+    *  This test creates an async delegate that is expected to trigger the response handler (responseHandler11).
+    *  That handler creates another async delegate, triggerring another response handler (responseHandler12).
+    *
+    *  Chaining is occurring in response handlers.
+    *
+    *  This test should always Succeed.
+    */
+    @AsyncTest
+    public function testResponseChainSuccess(factory:AsyncFactory) : Void
+    {
+        expectedValue = L1_RESPONS_FACTOR * L2_RESPONS_FACTOR;
+
+        h1 = factory.createHandler ( this, responseHandler11, 250, forbiddenTimeOutHandler );
+        Timer.delay ( h1, 10 );
+    }
+
+    public function responseHandler11() : Void
+    {
+        actualValue *= L1_RESPONS_FACTOR;
+
+        t1 = Async.handler ( this, responseHandler12, 250, forbiddenTimeOutHandler );
+        Timer.delay ( t1, 10 );
+    }
+
+    public function responseHandler12() : Void
+    {
+        actualValue *= L2_RESPONS_FACTOR;
+    }
+
+    /**
+    *  This test creates an async delegate that is expected to trigger the response handler (responseHandler21).
+    *  That (responseHandler21) creates another async delegate implemented to result in a time-out.
+    *  That time-out handler triggers an Assert.fail condition.
+    *
+    *  Chaining is occurring in response handlers.
+    *
+    *  This test is expected to be reported as FAILed with "Ignore: this is EXPECTED" as part of its error message.
+    */
+    @AsyncTest
+    public function testTimeoutHandlerFailure(factory:AsyncFactory) : Void
+    {
+        expectedValue = L1_RESPONS_FACTOR * FAILURE_TIMEOUT_FACTOR;
+
+        h2 = factory.createHandler ( this, responseHandler21, 250, forbiddenTimeOutHandler );
+        Timer.delay ( h2, 10 );
+    }
+
+    public function responseHandler21() : Void
+    {
+        actualValue *= L1_RESPONS_FACTOR;
+
+        t2 = Async.handler ( this, forbiddenResponseHandler, 250, failingTimeOutHandler );
```

Comment:
About the asymetry between `factory.createHandler` (explicit) and `Async.handler` (implicit): could we instead have a `factory.createAsyncHandler` which would be passed the `factory` reference?

Reply:
Oh I thought I removed the `Async.handler` cases. Your suggestion sounds like a good compromise

I think the goal (originally) was to allow switching from sync to async without having to add an additional argument for the factory, but that change broke backward compatibility and prevented the pull request from ever being merged. I'm fine with leaving the factory argument in to get some of the other fixes/improvements we need